src/topology/sheaves/locally_surjective.lean

```diff
@@ -0,0 +1,162 @@
+/-
+Copyright (c) 2022 Sam van Gool and Jake Levinson. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Sam van Gool, Jake Levinson
+-/
+
+import topology.sheaves.presheaf
+import topology.sheaves.stalks
+
+/-!
+
+Locally surjective maps of presheaves.
+
+Let X be a topological space, ℱ and 𝒢 presheaves on X, T : ℱ ⟶ 𝒢 a map.
+
+In this file we formulate two notions for what it means for
+T to be locally surjective:
+
+  1. For each open set U, each section t ∈ 𝒢(U) is in the image of T
+     after passing to some open cover of U.
+
+  2. For each x : X, the map of *stalks* Tₓ : ℱₓ ⟶ 𝒢ₓ is surjective.
+
+We prove that these are equivalent.
+
+-/
+
+universes v u
+
+noncomputable theory
+
+open category_theory
+open topological_space
+open opposite
+
+section locally_surjective
+
+/-- Let C be a concrete category, X a topological space. -/
+variables {C : Type u} [category.{v} C] [concrete_category.{v} C] {X : Top.{v}}
+
+/-- Let ℱ, 𝒢 : (opens X)ᵒᵖ ⥤ C be C-valued presheaves on X. -/
+variables {ℱ : X.presheaf C} {𝒢 : X.presheaf C}
+
+/-- When U is an open set, we introduce the notation "Γ_ ℱ U"
+for the set of sections of ℱ over U.
+-/
+@[nolint has_inhabited_instance] -- This may be empty.
+def sections_of_presheaf_on_open (ℱ : X.presheaf C) (U : opens X) : Type v :=
+  (forget C).obj (ℱ.obj (op U))
```

Comment:
I don't think you should mention `forget C` that often.
You should use `local attribute [instance] concrete_category.has_coe_to_fun concrete_category.has_coe_to_sort` instead.

Under this instance, I don't think most of the auxiliary definition and notations are nessecary.

Reply:
I now see that Adam has already proposed this on Zulip. But yes I also think this is the right way to go.

Reply:
OK, I will use Adam's proposed cleaner proof (https://gist.github.com/adamtopaz/802c0ad2491ef63e37b0d040d85e9eef).